### ISSUE TITLE
Search functionality consistently misses entries in 4.1.1

### ISSUE THREAD

**Issue by obakha** [Kunzisoft/KeePassDX#1946]
**Describe the bug**

After updating to 4.1.1, the search functionality has become significantly less effective. It now consistently misses out on entries that I'm sure exist in the database, even when using correct search terms.

Specific Example:

One specific example of this issue is when searching for my Google account. The login page for Google has the domain "www.accounts.google.com." When I search using the term "accounts," the search yields no results, despite the fact that the entry for my Google account should be easily found.

**To Reproduce**

1. Open KeePassDX 4.1.1.
2. Open a database that contains an entry for a Google account.
3. Tap the search icon in the top-right corner.
4. Enter the search term "accounts."
5. Observe that no results are returned.

**Expected behavior**

The search should return the entry for the Google account, since the domain of the login page contains the term "accounts."

**KeePassDX:**

 - Version: [4.1.1]
 - Build: [Free]
 - Language: [English]

**Android:**

 - Device: [Samsung A04]
 - Version: [10]

**Additional context**

This issue was not present in previous versions of KeePassDX. It only started occurring after updating to version 4.1.1.

**Comment by Starkstromkonsument** [comment]
It seems to be a problem with the trailing slash in the field URL. The search returns entries for URL = `https://example.org` but not for URL = `https://example.org/`. Can anyone confirm this?

**Comment by obakha** [comment]
No, I still get no results, even if an entry has no slach at the end, for example: "https://accounts.google.com" .

**Comment by J-Jamet** [comment]
There is indeed a problem with the algorithm with 4.1.0 and 4.1.1. I'll try to solve it properly.

**Comment by stefanhorning** [comment]
I  can confirm the problem is with the trailing slash.
